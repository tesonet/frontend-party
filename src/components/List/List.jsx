import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "reactstrap";
import "./List.scss";

const List = ({ headers, body }) => (
    <Table>
        <thead>
            <tr>
                {headers.map((header, index) => (
                    <th
                        style={{ textAlign: index > 0 && "right" }}
                        onClick={header.action}
                        key={index}
                    >
                        {header.name}
                    </th>
                ))}
            </tr>
        </thead>
        <tbody>
            {body.map((item, itemIndex) => (
                <tr key={itemIndex}>
                    {headers.map((header, headerIndex) => (
                        <td key={headerIndex} style={{ textAlign: headerIndex > 0 && "right" }}>
                            {item[header.key]}
                        </td>
                    ))}
                </tr>
            ))}
        </tbody>
    </Table>
);

export default List;

List.propTypes = {
    headers: PropTypes.array,
    body: PropTypes.array
};
