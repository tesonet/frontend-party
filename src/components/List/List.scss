th {
    cursor: pointer;
}

thead {
    background: #f5f5f5;
    color: #a8a8a8;
    th {
        font-weight: 400;
    }
}
