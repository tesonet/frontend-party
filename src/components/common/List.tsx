import React from 'react';
import styled from 'styled-components';
import Box from '../styled/Box';

type ListTypes = {
  children: React.ReactNode;
};

export const ListItem = styled.div``;

export const ListRow = styled.div`
  display: flex;
  justify-content: space-between;
  border-radius: ${props => props.theme.radii[0]};
  height: ${props => props.theme.sizes[0]};
  font-size: ${props => props.theme.fontSizes[1]};
  padding: 25px 15px 20px 15px;
  color: #666;
`;

export const ListHeaderRow = styled(ListRow)`
  background-color: ${props => props.theme.colors.listHeading};
  font-size: ${props => props.theme.fontSizes[0]};
  color: #999;
`;
export const ListScrollable = styled(ListRow)`
  background-color: ${props => props.theme.colors.listHeading};
  font-size: ${props => props.theme.fontSizes[0]};
`;

export default function List({ children }: ListTypes) {
  return <Box>{children}</Box>;
}
