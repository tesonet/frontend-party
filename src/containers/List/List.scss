.list-header {
    display: flex;
    justify-content: space-between;
    margin: 25px 10px;
}

.list-logo {
    height: 25px;
}

.list-logout {
    background-color: #fff !important;
    border: 1px solid #fff;
    transition: border-color 0.1s linear;
    
    .list-logout-icon {
        margin-right: 5px;
    }

    &:hover {
        border-color: #99cc33;
    }
}