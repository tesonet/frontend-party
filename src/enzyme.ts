import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
