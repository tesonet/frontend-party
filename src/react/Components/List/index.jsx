import {connect} from "react-redux";
import ListPage from "./listPage";

const mapStateToProps = (token) => {
   return{
       token: token
   }
};

export default connect(mapStateToProps, null)(ListPage);
