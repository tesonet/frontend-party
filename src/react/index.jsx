import React from 'react';
import ReactDOM from 'react-dom';
import store from './Store/';
import Router from './Router/';

ReactDOM.render(<Router store={store}/>, document.getElementById('react-root'));