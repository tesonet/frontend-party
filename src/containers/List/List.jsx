
import React, { useEffect, useState } from 'react';
import GetServers from '../../services/ServersService';
import Table from '../../components/Table/Table';
import logo from '../../assets/images/logo-dark.png'
import logout from '../../assets/icons/ico-logout.png'
import './List.scss';

const List = ({ history }) => {
    const token = localStorage.getItem('token');
    const [servers, setServers] = useState([]);

    const logOut = () => {
        localStorage.removeItem('token');
        history.push('/login');
    }

    useEffect(() => {
        GetServers(token).then(results => {
            setServers(results);
        })
    }, [token]);

    return (
        <div>
            <div className="list-header">
                <img alt="Logo" src={logo} className="list-logo" />
                <button type="button" onClick={logOut} className="list-logout">
                    <img alt="Logout" className="list-logout-icon" src={logout} />
                    Logout
                </button>
            </div>
            <Table servers={servers}></Table>
        </div>
    )
}

export default List;