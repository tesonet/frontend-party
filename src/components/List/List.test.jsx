import * as React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import List from "./List";

configure({ adapter: new Adapter() });

describe("List component", () => {
    const headers = [
        {
            name: "NAME",
            key: "name",
            action: () => {}
        },
        {
            name: "DISTANCE",
            key: "distance",
            action: () => {}
        }
    ];

    const body = [
        { name: "Lithuania", distance: "1" },
        { name: "Latvia", distance: "2" },
        { name: "Estonia", distance: "3" }
    ];

    it("should render component correctly", () => {
        const wrapper = shallow(<List headers={headers} body={body} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should receive 2 headers", () => {
        const wrapper = mount(<List headers={headers} body={body} />);
        expect(wrapper.props().headers).toHaveLength(headers.length);
    });

    it("should show 3 list items", () => {
        const wrapper = mount(<List headers={headers} body={body} />);
        expect(wrapper.props().body).toHaveLength(body.length);
    });
});
