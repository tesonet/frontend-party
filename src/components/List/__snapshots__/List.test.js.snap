// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render component correctly 1`] = `ShallowWrapper {}`;
